hype-submitter/src/main/java/com/spotify/hype/runner/LocalDockerRunner.java

```diff
@@ -0,0 +1,188 @@
+/*-
+ * -\-\-
+ * hype-submitter
+ * --
+ * Copyright (C) 2016 - 2017 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.hype.runner;
+
+import com.spotify.docker.client.DockerClient;
+import com.spotify.docker.client.exceptions.DockerException;
+import com.spotify.docker.client.messages.ContainerConfig;
+import com.spotify.docker.client.messages.ContainerCreation;
+import com.spotify.docker.client.messages.ContainerInfo;
+import com.spotify.docker.client.messages.HostConfig;
+import com.spotify.hype.model.RunEnvironment;
+import com.spotify.hype.model.StagedContinuation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+import static com.google.common.collect.ImmutableList.of;
+
+public class LocalDockerRunner implements DockerRunner {
+
+  private static final Logger LOG = LoggerFactory.getLogger(LocalDockerRunner.class);
+  private static final String GCLOUD_CREDENTIALS = "GOOGLE_APPLICATION_CREDENTIALS";
+  private static final String STAGING_VOLUME = "/staging";
+  private static final int POLL_PODS_INTERVAL_SECONDS = 5;
+
+  private final DockerClient client;
+  private final Boolean keepContainer;
+  private final Boolean keepTerminationLog;
+  private final Boolean keepVolumes;
+
+  public LocalDockerRunner(final DockerClient client,
+                           Boolean keepContainer,
+                           Boolean keepTerminationLog,
+                           Boolean keepVolumes) {
```

Comment:
Inconsistent final/non-final. Not a big deal, but we're mostly going with non-final for args, and final for local vars.